File: utils/location_utils.py
# utils/location_utils.py

from geopy.distance import geodesic
import json
from pathlib import Path

# BRANCHES = Path(__file__).parent.parent.parent / "data" / "branches.json"
BRANCHES = json.loads((Path(__file__).parent.parent / "data" / "branches.json").read_text())

def get_branch_from_location(latitude, longitude):
    user_coords = (float(latitude), float(longitude))
    print(BRANCHES)
    for branch_name, info in BRANCHES.items():
        coords = tuple(info["coordinates"])
        if geodesic(user_coords, coords).km <= 10:
            return branch_name
    return None

File: utils/locationUtils.js
const { getDistance } = require('geolib');
const branches = require('../data/branches.json');

const DEFAULT_RADIUS_KM = Number(process.env.BRANCH_RADIUS_KM || 3);

function findClosestBranch(latitude, longitude, radiusKm = DEFAULT_RADIUS_KM) {
  let closest = null;
  let minDistance = Infinity;
  for (const [name, info] of Object.entries(branches)) {
    const [branchLat, branchLon] = info.coordinates;
    const distance = getDistance(
      { latitude, longitude },
      { latitude: branchLat, longitude: branchLon }
    );
    if (distance < minDistance) {
      minDistance = distance;
      closest = { name, map_link: info.map_link, contact: info.contact, distance };
    }
  }
  if (closest && minDistance <= radiusKm * 1000) {
    return closest;
  }
  return null;
}

module.exports = { findClosestBranch };
